avm/src/vm/opcode.rs

```diff
@@ -1522,6 +1522,364 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
     Ok(None)
   });
 
+  // Saturating Arithmetic opcodes
+  cpu!(saddi8 => fn(args, hand_mem) {
+    let a = hand_mem.read_fixed(args[0])? as i8;
+    let b = hand_mem.read_fixed(args[1])? as i8;
+    if a > 0 && b > 0 && a > std::i8::MAX - b {
+      hand_mem.write_fixed(args[2], std::i8::MAX as i64)?;
+      return Ok(None);
+    }
+    if a < 0 && b < 0 && a < std::i8::MIN - b {
+      hand_mem.write_fixed(args[2], std::i8::MIN as i64)?;
+      return Ok(None);
+    }
+    hand_mem.write_fixed(args[2], (a + b) as i64)?;
+    Ok(None)
```

Comment:
why not use rust's [saturating add](https://doc.rust-lang.org/std/primitive.i8.html#method.saturating_add)?

Reply:
1. I forgot about it.
2. Particularly for division, we need to make sure it has the same semantics as here (true of all of them, but unlikely to differ for adding and subtracting)

Reply:
there's no saturating division implementation so I think it makes sense to maintain the manual implementation